### PR TITLE
CEDS-2658 upgrade scala version to 2.12

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -63,3 +64,14 @@ lazy val scoverageSettings: Seq[Setting[_]] = Seq(
...
+lazy val silencerSettings: Seq[Setting[_]] = {
+  val silencerVersion = "1.7.0"
+  Seq(
+    libraryDependencies ++= Seq(compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full)),
+    // silence all warnings on autogenerated files
+    scalacOptions += "-P:silencer:pathFilters=target/.*",
+    // Make sure you only exclude warnings for the project directories, i.e. make builds reproducible
+    scalacOptions += s"-P:silencer:sourceRoots=${baseDirectory.value.getCanonicalPath}"
+  )
+}
```
Comment (PatrykRudnicki) [inline build.sbt:77]: Is this necessary?

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -6,34 +6,34 @@ object AppDependencies {
...
   val compile: Seq[ModuleID] = Seq(
     ws,
-    "uk.gov.hmrc"          %% "play-conditional-form-mapping" % "1.2.0-play-26",
+    "uk.gov.hmrc"          %% "play-conditional-form-mapping" % "1.3.0-play-26",
     "uk.gov.hmrc"          %% "logback-json-logger"           % "4.8.0",
     "uk.gov.hmrc"          %% "govuk-template"                % "5.55.0-play-26",
     "uk.gov.hmrc"          %% "play-health"                   % "3.15.0-play-26",
     "uk.gov.hmrc"          %% "play-ui"                       % "8.11.0-play-26",
-    "uk.gov.hmrc"          %% "bootstrap-play-26"             % "1.13.0",
+    "uk.gov.hmrc"          %% "bootstrap-frontend-play-26"    % "2.24.0",
```
Comment (PatrykRudnicki) [inline project/AppDependencies.scala:14]: What's this? I think this library need to be without frontend

Comment (MaciejRewera) [reply project/AppDependencies.scala:14]: This is some new approach that has been taken: https://github.com/hmrc/bootstrap-play

File: build.sbt (hunk 1 above)
Comment (MaciejRewera) [reply build.sbt:77]: This is supressing (mostly invalid) warnings from twirl templates and auto-generated files. This seems to be a bug, with no intent to fix it: https://github.com/playframework/playframework/issues/7382

File: project/AppDependencies.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply project/AppDependencies.scala:14]: That's great, good to know

**Review by PatrykRudnicki — APPROVED** [review]
LGTM!

Please remember to update the artifact to 2.12 in service-manager-config

**Pull request merged**